packages/okta-adapter/src/filters/policy_rule_priority.ts

```diff
@@ -0,0 +1,252 @@
+}
+
+// For AccessPolicy, the priority index starts from 1, while for others it starts from 0.
+const setPriority = (typeName: string, priority: number): number => {
+  if (typeName === ACCESS_POLICY_TYPE_NAME) {
+    return priority
+  }
+  return priority + 1
+}
+
+const deployPriorityChange = async ({
+  client,
+  priority,
+  rule,
+  config,
+}: {
+  client: OktaClient
+  priority: number
+  rule: InstanceElement
+  config: OktaConfig
+}): Promise<void> => {
+  const { id } = rule.value
+  const { type } = rule.value
+  const policy = getParents(rule).find(parent => ALL_SUPPORTED_POLICY_NAMES.includes(parent.elemID.typeName))?.value
+  if (policy === undefined) {
+    throw new Error('Failed to deploy priority change due to missing policy')
+  }
+  const policyId = policy.value.id
+  const policyTypeName = policy.elemID.typeName
+  const data = { priority: setPriority(policyTypeName, priority), type, name: rule.value.name }
+  const typeDefinition = config.apiDefinitions.types[rule.elemID.typeName]
+  const deployRequest = typeDefinition.deployRequests ? typeDefinition.deployRequests.modify : undefined
+  const deployUrl = deployRequest?.url
+  if (deployUrl === undefined) {
+    throw new Error('Failed to deploy priority change due to missing url')
+  }
+  const url = createUrl({ instance: rule, url: deployUrl, additionalUrlVars: { ruleId: id, policyId } })
+
+  await client.put({ url, data })
+}
+
+/*
+ * Manages the priorities of policy rules within each Policy by generating an InstanceElement
+ * for the policy rules' priorities. Each priority instance contains the policy rules sorted
+ * by their priority, including the default rule. The default rule is always set to be last.
+ * In deployment, we deploy the priorities, not the policy rules themselves.
+ */
+const filter: FilterCreator = ({ config, client }) => ({
+  name: 'policyRulePrioritiesFilter',
+  onFetch: async elements => {
+    const instances = elements.filter(isInstanceElement)
+    const policiesRules = instances.filter(instance =>
+      ALL_SUPPORTED_POLICY_RULE_NAMES.includes(instance.elemID.typeName),
+    )
+    const policyAndrules = Object.values(
+      policiesRules.reduce(
+        (acc, rule) => {
+          // Autorization policy rules has more than one parent
+          const policy = getParents(rule).find(parent =>
+            ALL_SUPPORTED_POLICY_NAMES.includes(parent.elemID.typeName),
+          )?.value
+          if (policy === undefined) {
+            log.warn('Policy not found for rule %s', rule.elemID.getFullName())
+            return acc
+          }
+          const policyName = policy.elemID.getFullName()
+          if (!acc[policyName]) {
+            acc[policyName] = { policy, rules: [] }
+          }
+          acc[policyName].rules.push(rule)
+          return acc
+        },
+        {} as {
+          [key: string]: {
+            policy: InstanceElement
+            rules: InstanceElement[]
+          }
+        },
+      ),
+    )
```

Comment:
nit - personally i think reduce is sometimes difficult to read, and also I think we can easily avoid the casting and add using push here

I think this should work, but you can leave with reduce if its gets complicated (didn't include the entire logic)
```suggestion
        const policyAndRules = _(policiesRules).map(rule => {
      const parent = getParents(rule).find(isParentPolicy) // your code here...
      return parent ? ({ rule, parent }) : undefined
    })
    .filter(values.isDefined)
    .groupBy(({ parent }) => parent?.elemID.getFullName())
    .mapValues(v => v.map(({ rule }) => rule))
```